homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -37,194 +40,255 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class HeatMeterSensorEntityDescription(SensorEntityDescription):
+    """Heat Meter sensor description."""
+
+    value_fn: Callable | None = None
+    response_key: str | None = None
```

Comment:
Why is this needed? As far as i can follow, the only occasions where these differ from `key` is handled expicitly in
```py
         if self.key == "heat_usage":
...
         if self.key == "heat_previous_year":
```
It looks like using `key` after that would be fine too.
And you could remove `getattr` and `# type: ignore[attr-defined]` in L353

Reply:
In that case I'd make `entity_description.response_key` a required key. No `None` check in `native_value()` needed then.
You can use a dataclass-Mixin to achieve that. See eg. https://github.com/home-assistant/core/blob/d106cb48d2cebfc41883e9faf51b2769b8b5dffb/homeassistant/components/rdw/sensor.py#L28-L39

Reply:
You're amazing! This is exactly what I was looking for. Works perfectly, thanks!